internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepository.java

```diff
@@ -372,23 +392,37 @@ public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
 
   @Override
   public Optional<SchemaEntry> getSupersetSchema(String storeName) {
-    schemaLock.readLock().lock();
+    getSchemaLock().readLock().lock();
     try {
       fetchStoreSchemaIfNotInCache(storeName);
-      SchemaData schemaData = schemaMap.get(getZkStoreName(storeName));
+      SchemaData schemaData = getSchemaMap().get(getZkStoreName(storeName));
       if (schemaData == null) {
         throw new VeniceNoStoreException(storeName);
       }
       Optional<Integer> supersetSchemaID = getSupersetSchemaID(storeName);
```

Comment:
was wondering if we already have derived map updated with latest super-set schema's update schema and store change is still not propagated. in that case should we return the latest derived schema map even if store.supersetschmaID is state?

Reply:
I think given the fact that schema repo listens to store changed actively, the chance could be low. Even if your concern is valid, we could not use latest value schema or derived schema to deserialize the existing value or incoming partial update, as the latest one might have less field and could lead to deserialization failure or data missing as well.

Reply:
Super schema would contain all the fields and using the latest one should always work,  using a stale one from store config id may not work. But I agree, chances are less.

Reply:
Discussed offline and we think the chance in practice is low.